File: pgm/ABC_C#/ABC_145_B.cs
using System;
// using System.Collections.Generic;
// using System.Linq;
// using System.Text;

class Program
{
    static void Main()
    {
        int N = int.Parse(Console.ReadLine());
        string S = Console.ReadLine();
        Console.WriteLine(N % 2 == 0 ?
            S.Substring(0, N / 2) == S.Substring(N / 2) ? "Yes" : "No" : "No");
        return;
    }
}

File: pgm/ABC_Visual_Basic/ABC_145_B.vb
' Imports System.Collections.Generic
' Imports System.Text
' Imports System.Text.RegularExpressions

Class Main
    Public Shared Sub Main()
        Dim N As Integer = Console.ReadLine()
        Dim S As String = Console.ReadLine()
        Console.WriteLine(If(N Mod 2 = 0, _
            If(S.Substring(0, N / 2) = S.Substring(N / 2), "Yes", "No"), "No"))
        Return
    End Sub
End Class
